### ISSUE TITLE
[Bounty] Resolves issues with etherscan-verify for the ExchangeFactory.sol

### ISSUE THREAD

**Issue by 0xean** [ElasticSwap/elasticswap#79]
# Background

Currently when attempting to verify the ExchangeFactory.sol on etherscan using `goerli` our snowtrace using `fuji` the ExchangeFactory.sol fails to verify while all other contracts are able to be verified. 

```
$ npx hardhat --network fuji etherscan-verify --api-key <YOUR API KEY>
already verified: BaseToken (0x328684b9AcCB705E2782aC9b25458352cbAb0c81), skipping.
verifying ExchangeFactory (0xBe62889d7bbb98415123D76D3D862E4482463e52) ...
waiting for result...
Failed to verify contract ExchangeFactory: NOTOK, Fail - Unable to verify
{
  "apikey": "XXXXXX",
  "module": "contract",
  "action": "verifysourcecode",
  "contractaddress": "0xBe62889d7bbb98415123D76D3D862E4482463e52",
  "sourceCode": "...",
  "codeformat": "solidity-standard-json-input",
  "contractname": "src/contracts/ExchangeFactory.sol:ExchangeFactory",
  "compilerversion": "v0.8.4+commit.c7e474f2",
  "constructorArguements": "0000000000000000000000005fd46dbfceba8eb28485ff7733fc7c00ca861d7c",
  "licenseType": 5
}
```

# How to reproduce
1. Check out repo
2. install dependencies
3. get a free api key from snowtrace
4. run `npx hardhat --network fuji etherscan-verify --api-key <YOUR API KEY>`
5. See that you are unable to verify the contract deployed [here](https://testnet.snowtrace.io/address/0xBe62889d7bbb98415123D76D3D862E4482463e52)

# Acceptance Criteria for bounty payout
1. https://testnet.snowtrace.io/address/0xBe62889d7bbb98415123D76D3D862E4482463e52 shows a verified contract
2. We have a repeatable way to verify the ExchangeFactory.sol on all networks when we deploy

# Bounty payout
Once confirmed the bounty hunter will receive 40 TIC Tokens from the ElasticSwap Treasury.

**Comment by 0xean** [comment]
Also worth noting that the suggestion from hardhat

```
Etherscan sometime fails to verify when only metadata sources are given. See https://github.com/ethereum/solidity/issues/9573. You can add the option --solc-input to try with full solc-input sources. This will include all contract source in the etherscan result, even the one not relevant to the contract being verified
```

also fails from my attempts.

**Comment by millicare** [comment]
I resolved that problem just by using `npx hardhat verify ` command instead of using `npx hardhat etherscan-verify`
`npx hardhat verify --network fuji <CONTRACT_ADDRESS> <ARGS>`
(It is able to make this verify command running by making script. to detect all deployed contracts and verify them all at a once.)
Ahead of running this command, there should be etherscan config set in hardhat.config.js
```
etherscan: {
    // Your API key for Snowtrace
    // Obtain one at https://snowtrace.io/
    apiKey: process.env.ETHERSCAN_API_KEY,
  },
```
hope it could be any of help to close this ticket. 😉

**Comment by 0xean** [comment]
resolved in #81
